CMakeLists.txt

```diff
@@ -0,0 +1,362 @@
+
+include(CheckSymbolExists)
+check_symbol_exists(fdatasync "unistd.h" HAVE_FDATASYNC)
+
+configure_file(
+  "${PROJECT_SOURCE_DIR}/port/port_config.h.in"
+  "${PROJECT_BINARY_DIR}/include/port/port_config.h"
+)
+
+include_directories(
+  "${PROJECT_BINARY_DIR}/include"
+  "${PROJECT_SOURCE_DIR}"
+)
+
+# POSIX code is specified separately so we can leave it out in the future.
+add_library(leveldb_port_posix OBJECT port/port_posix.cc
+    ${PROJECT_BINARY_DIR}/include/port/port_config.h
+    port/atomic_pointer.h
+    port/port_posix.h
+    port/port.h
+)
+
+add_library(leveldb ""
+  $<TARGET_OBJECTS:leveldb_port_posix>
+  "db/builder.cc"
+  "db/builder.h"
+  "db/c.cc"
+  "db/db_impl.cc"
+  "db/db_impl.h"
+  "db/db_iter.cc"
+  "db/db_iter.h"
+  "db/dbformat.cc"
+  "db/dbformat.h"
+  "db/dumpfile.cc"
+  "db/filename.cc"
+  "db/filename.h"
+  "db/log_format.h"
+  "db/log_reader.cc"
+  "db/log_reader.h"
+  "db/log_writer.cc"
+  "db/log_writer.h"
+  "db/memtable.cc"
+  "db/memtable.h"
+  "db/repair.cc"
+  "db/skiplist.h"
+  "db/snapshot.h"
+  "db/table_cache.cc"
+  "db/table_cache.h"
+  "db/version_edit.cc"
+  "db/version_edit.h"
+  "db/version_set.cc"
+  "db/version_set.h"
+  "db/write_batch_internal.h"
+  "db/write_batch.cc"
+  "port/port.h"
+  "port/thread_annotations.h"
+  "table/block_builder.cc"
+  "table/block_builder.h"
+  "table/block.cc"
+  "table/block.h"
+  "table/filter_block.cc"
+  "table/filter_block.h"
+  "table/format.cc"
+  "table/format.h"
+  "table/iterator_wrapper.h"
+  "table/iterator.cc"
+  "table/merger.cc"
+  "table/merger.h"
+  "table/table_builder.cc"
+  "table/table.cc"
+  "table/two_level_iterator.cc"
+  "table/two_level_iterator.h"
+  "util/arena.cc"
+  "util/arena.h"
+  "util/bloom.cc"
+  "util/cache.cc"
+  "util/coding.cc"
+  "util/coding.h"
+  "util/comparator.cc"
+  "util/crc32c.cc"
+  "util/crc32c.h"
+  "util/env_posix.cc"
+  "util/env.cc"
+  "util/filter_policy.cc"
+  "util/hash.cc"
+  "util/hash.h"
+  "util/logging.cc"
+  "util/logging.h"
+  "util/mutexlock.h"
+  "util/options.cc"
+  "util/posix_logger.h"
+  "util/random.h"
+  "util/status.cc"
+  "include/leveldb/c.h"
+  "include/leveldb/cache.h"
+  "include/leveldb/comparator.h"
+  "include/leveldb/db.h"
+  "include/leveldb/dumpfile.h"
+  "include/leveldb/env.h"
+  "include/leveldb/filter_policy.h"
+  "include/leveldb/iterator.h"
+  "include/leveldb/options.h"
+  "include/leveldb/slice.h"
+  "include/leveldb/status.h"
+  "include/leveldb/table_builder.h"
+  "include/leveldb/table.h"
+  "include/leveldb/write_batch.h"
+)
+target_include_directories(leveldb
+  PUBLIC
+    $<BUILD_INTERFACE:${PROJECT_SOURCE_DIR}/include>
+)
+target_compile_definitions(leveldb
+  PRIVATE
+    LEVELDB_PLATFORM_POSIX
+)
+
+# TODO(pwnall): This is only needed for port_posix.
+set(THREADS_PREFER_PTHREAD_FLAG ON)
+find_package(Threads REQUIRED)
+target_link_libraries(leveldb Threads::Threads)
+set(deps Threads)
+
+if(LEVELDB_WITH_CRC32C)
+  target_link_libraries(leveldb crc32c::crc32c)
```

Comment:
I guess it's public dependency, should be `target_link_libraires(... PUBLIC ...)`.